guides/micronaut-microservices-services-discover-eureka/bookinventory/java/src/main/java/example/micronaut/BookInventory.java

```diff
@@ -1,26 +1,24 @@
 package example.micronaut;
 
+import io.micronaut.core.annotation.Creator;
 import io.micronaut.core.annotation.Introspected;
 import io.micronaut.core.annotation.NonNull;
 
 import javax.validation.constraints.NotBlank;
-import javax.validation.constraints.NotNull;
 import java.util.Objects;
 
 @Introspected
 public class BookInventory {
 
     @NonNull
     @NotBlank
-    private String isbn;
+    private final String isbn;
 
-    @NonNull
-    @NotNull
-    private Integer stock;
-
-    public BookInventory() {}
+    private final int stock;
 
-    public BookInventory(@NonNull @NotBlank String isbn, @NonNull @NotNull Integer stock) {
+    @Creator
```

Comment:
I don't think `@Creator` is necessary. Am I missing anything?

If you want to specify it to educate the user about `@Creator` I think we should add a callout with a callout snippet.